### PR TITLE
Basic monitoring with Prometheus operator

### PR DESCRIPTION
**Pull request by bittrance**
This PR introduces the Prometheus operator and instruments Dkron and Traefik for scraping. Since this project is a learning exercise we pull in the operator through the semi-official Helm chart. See https://github.com/prometheus-operator/prometheus-operator/ for more information. We stick with Terraform for now, so we need the "alpha" version of the Terraform Kubernetes provider as it has support for arbitrary manifests, specifically CRDs. 

At the end, pretty graphs:

image

### CODE REVIEW

File: krony-app/dkron-api.tf [hunk 1]
```diff
@@ -38,6 +42,35 @@ resource "kubernetes_service" "dkron" {
   }
 }
 
+resource "kubernetes_manifest" "dkron_metrics" {
```
Comment (simongottschlag) [inline krony-app/dkron-api.tf]: An alternative to the alpha provider can be to use a local helm chart: https://github.com/XenitAB/terraform-modules/tree/main/modules/kubernetes/external-dns/charts/external-dns-extras

**Review by simongottschlag — COMMENTED** [review]
Cool project! 😊👍

**Comment by bittrance** [comment]
Speculative plan failed for this PR, probably because of https://github.com/hashicorp/terraform-provider-kubernetes-alpha/issues/168 . Applying this is staging worked fine, but then the Helm chart was applied before the cursom resource. May have to split the PR, but will wait for feedback in linked issue.

File: krony-app/dkron-api.tf (hunk 1 above)
Comment (bittrance) [reply krony-app/dkron-api.tf]: Well, the issue here turns out to be thst Terraform cannot at present apply CRDs and their CRs in the same run. That is a fairly serious limitation given the proliferation of CRDs in Kubernetes these days.

**Comment by bittrance** [comment]
I split this PR in two so as to merge only the Helm chart and not the CRD users. This will not really solve anything, but will answer https://github.com/hashicorp/terraform-provider-kubernetes-alpha/issues/168#issuecomment-808075407 .

**Pull request merged**